Add Newsletter Modal

whitelionx opened the pull request:
Resolves #931 
Account number: fb84a32a698a03dd327c4ce673f19fa0acf6e68cb50f143d5d7c080471bc4625

The modal is done, the only thing missing is the endpoint for saving the email in the database for the newsletter.

thesanjeevsharma commented on src/components/Footer/Footer.scss:
```diff
@@ -21,6 +21,11 @@
 
+  &__newsletter-form {
+    margin-top: 42px;
+    color: var(--color-white);
```
Alphabetical order in css - color should come first!

thesanjeevsharma commented on src/components/Footer/index.tsx:
```diff
@@ -6,6 +6,7 @@ import Logo from 'assets/svgs/thenewboston-white.svg';
 import SocialMediaIcon from 'components/SocialMediaIcon';
 import {SocialMedia} from 'types/social-media';
 
+import NewsletterForm from 'components/Newsletter/NewsletterForm';
```
Don't import it like this. Export and Import from components/index.js

thesanjeevsharma commented on src/components/Newsletter/NewsletterForm/index.scss:
```diff
@@ -0,0 +1,28 @@
+
+  input {
+    width: 240px;
+    padding: 8px;
```
Alphabetical order

thesanjeevsharma commented on src/components/Newsletter/NewsletterForm/index.tsx:
```diff
@@ -0,0 +1,26 @@
+import React, {FC} from 'react';
+import './index.scss';
```
Name it Newsletter.scss instead of index.scss

thesanjeevsharma commented on src/components/Newsletter/NewsletterForm/index.tsx:
```diff
@@ -0,0 +1,26 @@
+    <form
+      className="newsletter-form"
+      onSubmit={(e) => {
+        // TODO - Logic for saving the email for the upcoming newsletter
```
Can you complete the flow for it? Write the API calling fucntion. So, before merging I'll just replace the URL.

thesanjeevsharma commented on src/components/Newsletter/NewsletterForm/index.tsx:
```diff
@@ -0,0 +1,26 @@
+        console.log('Submitted');
+      }}
+    >
+      <input type="text" className="Input" placeholder="Email address" />
```
Make it a controlled input and add some validations to it. Use yup for validations.

thesanjeevsharma commented on src/components/Newsletter/index.scss:
```diff
@@ -0,0 +1,40 @@
+.newsletter-modal {
+  position: fixed;
```
Alphabetical order in this file.

thesanjeevsharma commented on src/components/Newsletter/index.tsx:
```diff
@@ -0,0 +1,33 @@
+import React, {FC, Fragment, useState, useEffect} from 'react';
+import {Icon, IconType, Shadow} from 'components';
+import NewsletterForm from './NewsletterForm';
+import './index.scss';
```
Newsletter.scss

thesanjeevsharma commented on src/components/Newsletter/index.tsx:
```diff
@@ -0,0 +1,33 @@
+import './index.scss';
+
+const NewsletterModal: FC = () => {
+  const [modal, setModal] = useState(false);
```
Add missing type for this state

whitelionx commented:
Got it @thesanjeevsharma now I know what the rules are for the PR's

whitelionx commented:
Just to be sure do you guys want that I insert Formik into it with yup or normal form? :D @thesanjeevsharma

Hristijan95 commented:
@whitelionx , use Formik with yup. You can take the Formik component for input from the Account-Manager repo.

whitelionx commented:
Added Yup & Formik ( I used the useFormik hook as it is cleaner ), made the css alphabetical, adjusted things to be responsive, the only thing remaining is the endpoint. @thesanjeevsharma @Hristijan95

whitelionx commented:
Just to add to it yesterday I added an error message when the validation kicks in, I hope that's okay?

lapstjup commented:
Hey @whitelionx I can see the #585 PR active again. There are a lot of form components and elements that are coming from AM to Website in that PR. So once that is done, can you reuse all that is required to re-create this news letter modal ? 
I am pretty sure you will find custom Input and Modal components once that PR gets merged.

whitelionx commented:
This isn't a complicated form, this is just a single input field I made it as a form even though I didn't need to, so I don't see many issues keeping it this way @lapstjup but later on when all stuff gets merged it is easy to exchange the certain element for the component. No need to wait for the one or another.

thesanjeevsharma commented:
@whitelionx I have merged #585. You can now use Form components and elements. Can you update your PR to use those? Also, resolve these conflicts.

whitelionx commented:
Yeah will update it and resolve the conflicts @thesanjeevsharma

buckyroberts commented:
@whitelionx This PR is not needed. We are going to have GitHub auth on our website soon so no email forms needed. My mistake for not prioritizing engineering tasks as well as I could have. Paid you the task reward out of my personal funds.
